ecl2df/summary.py

```diff
@@ -287,6 +287,72 @@ def df(
     return dframe
 
 
+def _fix_dframe_for_libecl(dframe: pd.DataFrame) -> pd.DataFrame:
+    """Fix a dataframe making it ready for EclSum.from_pandas()
+
+    * Ensures that the index is always datetime, and sorted.
+    * Removes BLOCK vectors, these are currently not supported as
+      it requires knowledge of the grid dimensions. Warnings
+      will be emitted for skipped columns
+
+    Args:
+        dframe (pd.DataFrame): Dataframe to read. Will not be modified.
+
+    Returns:
+        pd.DataFrame: Modified copy of incoming dataframe.
+    """
+    dframe = dframe.copy()
+    if "DATE" in dframe.columns:
+        dframe["DATE"] = pd.to_datetime(dframe["DATE"])
+        dframe = dframe.set_index("DATE", drop=True)
+    dframe.sort_index(axis=0, inplace=True)
```

Comment:
If "Date" was not in columns - check that the index is in fact of a date/datetime type and raise error if not?

Reply:
Now done.